### ISSUE TITLE
Version compatibility level higher in source than target prompt

### ISSUE THREAD

**Issue by maturmel** [microsoft/Analysis-Services#210]
Hi,

Would there be a way to auto-accept the prompt about version compatibility level being higher in source than target model? I am using  BismNormalizer.exe in a Azure DevOps pipeline to deploy my model changes and, of course the prompt cannot be answered interactively. I can work it out manually for the (very) few times it happens, but I feel like having an option either a command-line switch or a tag in the BSMN file structure would help towards automating the answer to this prompt.

Thanks and keep it up, Power BI and ALM Toolkit are awesome tools.
Marc

**Comment by christianwade** [comment]
Thanks for the feedback Marc. I'll add it to the to-do list to look at when I get the chance.

**Comment by ma-turmel** [comment]
Hello Christian,

I had some time to have a look through the code a little bit around this issue, one thing that came up to me was that my scenario described above, i.e. using BismNormalizer.exe command-line, should not be considered interactive mode for the comparison code. However I could not find anywhere in the code path that sets the Interactive properly to false, and as it is defaulted to true, hence the MessageBox will be invoked and block execution even in unattended/command-line scenarios. If there'd be a way to have that Interactive property set to false in my scenario, I would at least get the error message and the execution wouldn't block, then I could take manual action to resolve the compatibility version issue and rerun my Azure DevOps pipeline.

I'll work out a Pull Request that would enable these scenarios (interactive=false & a new "force update of target model version" switch) to be addressed.

**Comment by christianwade** [comment]
Hi Marc, please take a look at the command line options discussed from page 31 of the doc below as I think it allows parameterization for non-interactive mode when creds are provided.

https://github.com/microsoft/Analysis-Services/blob/master/AlmToolkit/Model%20Comparison%20and%20Merging%20for%20Analysis%20Services.pdf

**Comment by ma-turmel** [comment]
Hi Christian,

Indeed I had initially followed instructions from that page to invoke BismNormalizer.exe from command line, however it still invokes a windows Message Box in the case of different compat level between source and target. I can easily replicate this behavior by running the exe on a command prompt on my machine. There does not seem to be any place in the code that will set the "interactive" flag to false, unfortunately. The flag is not serialized to the BSMN file nor interfaced directly in command-line options. CredsProvided=true does not seem to set Interactive to false, at least from my experience.

Let me know if you have other ideas to work around the limitation. Thank you for your support !

**Comment by christianwade** [comment]
You were right. The command line shell may have been redone a while ago and didn't set Interactive=False. So that was a regression.

Can you try this pre-release build?
https://github.com/microsoft/Analysis-Services/releases/tag/5.0.68-PreRelease

In addition to the Interactive=False fix, it has the following new command line parameter:
/UpgradeCompatLevel:True|False : Automatically upgrade target compat level if it's less than the source one.

The command line exe is now BismNormalizer.CommandLine.exe. Hopefully that's OK.

I don't plan to do a full release for some time, but hopefully this will help.

**Comment by ma-turmel** [comment]
I'll gladly give it a shot, I will give you full feedback early next week. Thank you for taking the time to make this build, I appreciate it.

**Comment by christianwade** [comment]
Can you let me know when you have downloaded the MSI? I'd like to remove that pre-release when you have taken it.

**Comment by ma-turmel** [comment]
I've got the file on my machine, you may remove the link.

**Comment by christianwade** [comment]
Hi Marc, I found an issue where if you perform a compat level upgrade, TOM needs to resync schema to issues when do validation. It could affect your scenario, so please take another download from the pre-release below and let me know when you've done so, so I can remove it.

https://github.com/microsoft/Analysis-Services/releases/tag/5.0.69-PreRelease

**Comment by ma-turmel** [comment]
Hi Christian, thank you again, I've downloaded the new pre-release, you can remove it.

**Comment by ma-turmel** [comment]
I've done a few test scenarios with the pre-release build, it works as expected:
- Using the same command as I did before (except the .exe filename which changed), I now get an error message output about the version incompatibility, and the program exits. That should work fine in a ADO pipeline scenario;
- I also tested the new parameter /UpgradeCompatLevel:True and it did work also and upgraded the target. The next compare did not trigger the message, as expected, because the upgrade compat level happens immediately on the target.

I just need to integrate the new EXE "portable" version into my ADO pipelines, which should not be much work.

Thank you so much again for your help.
All the best for 2024!
